src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -74,6 +74,9 @@ public class BlockStoreQueue : IBlockStoreQueue
         /// <summary>Protects the batch from being modifying while <see cref="GetBlockAsync"/> method is using the batch.</summary>
         private readonly object getBlockLock;
 
+        /// <summary>Represents all blocks currently in the queue & pending batch, so that <see cref="GetBlockAsync"/> is able to return a value directly after enqueuing.</summary>
+        private Dictionary<uint256, ChainedHeaderBlock> blocksBeingPersisted;
```

Comment:
rename to blocksBeingProcessed. Because not all the blocks in the batch will be persisted. Sometimes we can have a reorg inside a batch so some of the blocks will be removed without even touching the disk

Reply:
Perhaps in that case just call it `pendingBlocks`